ci.py

```diff
@@ -0,0 +1,200 @@
+
+    def __init__(self):
+        self.global_branch_name = None
+        self.config = {}
+        self.custom_config = {}
+        self.read_custom_config()
+        self.read_config()
+        self.deps = self.config['deps']
+
+    def read_custom_config(self):
+        with open(YAML_CONFIG_CUSTOM, 'r') as f:
+            self.custom_config = yaml.load(f, Loader=yaml.FullLoader)
+
+    def write_custom_config(self, config):
+        with open(YAML_CONFIG_CUSTOM, 'w') as f:
+            yaml.dump(config, f, sort_keys=False)
+
+
+    def read_config(self):
+        with open(YAML_CONFIG, 'r') as f:
+            self.config = yaml.load(f, Loader=yaml.FullLoader)
+
+        if self.custom_config is not None:
+            # first we want to find global branch
+            for conf in self.custom_config['deps']:
+                if conf['name'] == 'global':
+                    self.global_branch_name = conf['branch']
+                    self.set_global_branches()
+                    break
+
+            # Yep we have high complexity here but list is short
+            for conf in self.custom_config['deps']:
+                if conf['name'] == 'global':
+                    continue
+                for dep in self.config['deps']:
+                    if dep['name'] == conf['name']:
+                        # TODO add support for other fields
+                        dep['branch'] = conf['branch']
+                        break
+                else:
+                    logging.error(f'Can"t find {conf["name"]} repo. We have list of repos in ci-default.yml')
+                    sys.exit(1)
+
+    def set_global_branches(self):
+        for dep in self.config['deps']:
+            url = dep['url']
+            g = git.cmd.Git()
+            # TODO maybe it'll be faster to use local data
+            output = g.ls_remote("--heads", url, self.global_branch_name)
+            if self.global_branch_name in output:
+                logging.info(f'Use branch {self.global_branch_name} for {dep["name"]}')
+                dep['branch'] = self.global_branch_name
+
+    def create_fb(self, branch_name):
+        import git
+        repo = git.Repo('.')
+
+        git = repo.git
+        for ref in repo.references:
+            if branch_name == ref.name:
+                git.checkout(branch_name)
+                break
+        else:
+            git.checkout('HEAD', b=branch_name)
+
+        if self.custom_config is not None:
+            for dep in self.custom_config['deps']:
+                if dep['name'] == 'global':
+                    dep['branch'] = branch_name
+        else:
+            global_branch = {'name': 'global', 'branch': branch_name}
+            self.custom_config = {'deps': [global_branch,]}
+
+        self.write_custom_config(self.custom_config)
+        repo.git.add(all=True)
+        repo.index.commit('Create fuature build')
+        origin = repo.remote(name='origin')
```

Comment:
this command doesn't push changes to remote, so you should push first before creating PR.

Reply:
Yep, I just deleted it when tested.